lib/observable-component.js

```diff
@@ -92,7 +92,12 @@ function updateObservableWithProps(observable, newProps, observer) {
 			}
 
 			if (type === 'splice') {
-				canReflect.splice(canKey.get(observable, key), values.index, values.deleteCount, values.insert);
+				if (Array.isArray(observable[key])) {
```

Comment:
Should probably use canKey.get here like the next line does, but cache the value so we don't have to look twice.